util/curl.go

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package util
+
+import (
+	"os/exec"
+	"bytes"
+	"fmt"
+)
+
+const (
+	defaultCurlCli = "/usr/bin/curl"
+)
+
+// Executes curl command with provided header and params.
+func CurlCommand(cli string, header string, url string, extraArgs ...string) {
```

Comment:
Can we change this to just take the token so we don't have to use BuildHeader every time before invoking this?

Reply:
I think it's better to have "header" as the parameter here for several reasons:
1. The logic to construct the authorization header from "token type" and "access token" has nothing to with the curl command itself, which is generic. The curl util is more flexible in its current form.
2. The SSO path does not give us a "token object" to work with, and uses BuildHeader differently. It is cleaner to pass the final "header" into the CurlCommand function.